controllers/cloud.redhat.com/namespacereservation_controller.go

```diff
@@ -187,6 +193,19 @@ func (r *NamespaceReservationReconciler) Reconcile(ctx context.Context, req ctrl
 			return ctrl.Result{}, err
 		}
 
+		duration, err := parseDurationTime(*res.Spec.Duration)
+		if err != nil {
+			r.Log.Error(err, "Cannot parse duration")
+			return ctrl.Result{}, err
+		}
+
+		averageRequestedDurationMetrics.With(prometheus.Labels{"controller": "namespacereservation"}).Observe(float64(duration.Hours()))
+
+		timer_res_to_deployment_end := time.Now()
+		elapsed := timer_res_to_deployment_start.Sub(timer_res_to_deployment_end)
```

Comment:
If this is supposed to measure how long it takes from reservation til fulfilment, we should use the createdTimestamp on the resource as the start time. Else multiple reconciliations will yield a breaking up of the time.

Reply:
DIdn't think about this- I'll update that now!